packages/sanity/src/core/util/draftUtils.ts

```diff
@@ -74,26 +83,40 @@ export function getIdPair(
     draftId: getDraftId(id),
     ...(version
       ? {
-          versionId: id.startsWith(`${version}.`) ? id : [version, getPublishedId(id)].join('.'),
+          versionId: isVersionId(id) ? id : getVersionId(id, version),
         }
       : {}),
   }
 }
 
 /** @internal */
 export function isPublishedId(id: string): id is PublishedId {
-  return !isDraftId(id)
+  return !isDraftId(id) && !isVersionId(id)
 }
 
 /** @internal */
 export function getDraftId(id: string): DraftId {
   return isDraftId(id) ? id : ((DRAFTS_PREFIX + id) as DraftId)
 }
 
+/**  @internal */
+export function getVersionId(id: string, bundle: string): string {
+  if (isVersionId(id)) {
+    const [version, bundleName, ...publishedId] = id.split('.')
```

Comment:
COULD: improve the naming here, to clarify that `version` is 'version'
```suggestion
    const [_versionPrefix, bundleName, ...publishedId] = id.split('.')
```